packages/integrations/components/pico/index.js

```diff
@@ -83,21 +86,16 @@ const Pico = (props) => {
     const widgetContainer = document.getElementById('pico-widget-container');
 
     if (widgetContainer) {
-      // Prevent the widget script from being added more than once.
-      setPicoInitialized(true);
-
       // Ensure the target nodes are only mounted once on the initial server load.
-      if (
-        ! document.getElementById('PicoSignal-container') &&
-        ! document.getElementById('PicoRule-button') &&
-        ! picoLoaded
-      ) {
+      if (! isPicoMounted()) {
         mountPicoNodes();
+        // Dispatch initial Pico page visit.
+        dispatchUpdatePicoPageInfo(picoPageInfo);
```

Comment:
This should dispatch the initial Page action, which isn't getting fired on initial render.